### ISSUE TITLE
Section items not dynamically updating

### ISSUE THREAD

**Issue by gorbypark** [andrew-levy/swiftui-react-native#53]
I have a fairly straightforward list, the data is coming from react-query, and when the data changes, I'm seeing weird behaviour, namely if the length of data changes (eg: something was deleted) there will be a blank "ItemWithArrow" component.  However, I added a little indicator of the length (`total address: {data.length}`) and that updates as expected.  If I force the entire section to rerender by doing `<Section key={data.length}>` then it works as expected.  However this isn't really a solution since I see the same behaviour when updating an item, there's just a blank item instead of the updated values.

I've tried replacing the `<ItemWithArrow>` component (which is just a button and label from swiftui-react-native) with just straight up Text (from react-native and from swiftui-react-native) with the same results.  Also, I've tried swapping out ForEach for a regular javascript map, so I don't believe it's that either.

Anyways, after lots of trial and error it seems that the Section component does not re-render (which is kinda expected in react) when the children do, however it's required for it to work.  I assume this is something to do with how it's bridging data to the native side, but I am not sure.

Adding a key to `<Section>`that loops through and concats some data to ensure it rerenders when anything changes is a workaround although not very pretty.

```tsx
         key={`shipping-addresses-${data
            .map((address) => `${address.id}-${address.addressLine1}`)
            .join("-")}`}
```

```tsx
<List> 
  <Section header="Shipping Addresses">
    {ForEach(data, (address, index) => (
      <ItemWithArrow
        key={index}
        title={address.addressLine1}
        action={() =>
          router.push({
            pathname: "/tabs/account/address/[address_id]",
            params: {
              address_id: address.id,
            },
          })
        }
      />
    ))}
    {!data.length && <Text>No saved addresses</Text>}
    <Text>Total addresses: {data.length}</Text>
    <Button action={() => router.push("/tabs/account/address/new")} title="Add address" />
  </Section>
</List>
 ```

**Comment by andrew-levy** [comment]
Thanks for reporting this! I have a fix for this, but im going to test it a little more. Will keep you updated @gorbypark

**Comment by andrew-levy** [comment]
@gorbypark This should be fixed in #54. Can you please try with `swiftui-react-native@6.3.2` and let me know if that resolves your issue?

**Comment by gorbypark** [comment]
Unfortunately it doesn't seem to be working, just upgraded to 6.3.2 and did a full expo prebuild --clean and build.

Here's the issue (no key on section), you can see updating the content works fine, but changing the number of items results in a blank spot:

https://github.com/user-attachments/assets/7b86ac5b-18b5-4581-86e1-ed41f3140fc7

And with a key forcing a rerender:

https://github.com/user-attachments/assets/7317f189-391a-4506-8224-886bb8810fe4

**Comment by andrew-levy** [comment]
@gorbypark  I tried creating an example that matches your setup here, I think I got close enough but lmk if you see a major difference. Here's what I have:

- I have some Context that stores an array of items
- On screen 1, I have a list of the items 
- On screen 2, I have buttons to add and delete from that list
- When navigating back to screen 1, I see the updated list with no blank rows.
- Not using any keys and it works

https://github.com/user-attachments/assets/b1bf26a3-f501-4737-932d-9f654386fe38

**Comment by gorbypark** [comment]
Well, well...it seems I wasn't on the latest version after all?  

I decided since I'm just working in a testing repo I would share it here. I stripped down the screen to the bare minimum, I created another list component from stock react native components and using the same query to show it was working there and not when using Section.  I swapped over the API side of things to sqlite so it would be easy for you to run.  I pushed it to GitHub and typed out a few instructions on how to start it. Then I decided I'd actually follow my instructions to make sure I didn't forget a step so I cloned the project, followed my steps...and it works!  

I suppose even though I did a full `npx expo prebuild --clean && yarn ios` after upgrading to 6.3.2 I was still running the old version somehow.  It's possible I bumped the version number but forgot the run yarn to actually install it 😅 

Anyways, thanks for the amazing library and quick turn around on this bug!
